go-controller/pkg/ovn/pods.go

```diff
@@ -374,29 +376,40 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) (err error) {
 			return fmt.Errorf("unable to create LSPSetDynamicAddresses command for port: %s", portName)
 		}
 		cmds = append(cmds, cmd)
-	} else {
+
+		// ensure we have/can allocate the IP in the annotation
+		if err = oc.lsManager.AllocateIPs(logicalSwitch, podIfAddrs); err != nil && err != ipallocator.ErrAllocated {
```

Comment:
Won't `podIfAddrs` always be empty here? This should be after `oc.getPortAddresses`?

Reply:
podIfAddrs are set to the annotation IPs. If we get a retry on the pod it will already have its annotation set

Reply:
right, but this is the `err != nil` block; can `err` be non-nil and `podIfAddrs` be set?

Reply:
aha, the diff made the indentation level unclear. nevermind, sorry.